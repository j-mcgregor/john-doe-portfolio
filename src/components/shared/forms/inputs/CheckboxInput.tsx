import React from 'react'

export interface CheckboxInputProps {
    label: string
    name: string
    isChecked: boolean
    onChange(e: React.ChangeEvent<HTMLInputElement>): void
}

const CheckboxInput: React.FC<CheckboxInputProps> = ({
    label,
    name,
    isChecked,
    onChange,
}) => {
    return (
        <div className="form-group flex flex-row">
            <input
                type="checkbox"
                data-testid={name}
                name={name}
                checked={isChecked}
                onChange={onChange}
                aria-label={name}
            />
            {label && <label htmlFor={name}>{label}</label>}
        </div>
    )
}

export default CheckboxInput
