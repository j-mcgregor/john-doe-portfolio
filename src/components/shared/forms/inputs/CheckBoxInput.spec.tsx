import React from 'react'
import { render } from '@testing-library/react'
import CheckboxInput from './CheckboxInput'

describe('CheckboxInput', () => {
    it('should render a button', () => {
        const mockFn = jest.fn()
        const utils = render(
            <CheckboxInput
                label="Test Checkbox"
                name="Test"
                isChecked={false}
                onChange={mockFn}
            />
        )

        expect(utils.baseElement).toMatchSnapshot()
    })
})
