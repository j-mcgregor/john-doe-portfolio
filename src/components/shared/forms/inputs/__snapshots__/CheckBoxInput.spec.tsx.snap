// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxInput should render a button 1`] = `
<body>
  <div>
    <div
      class="form-group flex flex-row"
    >
      <input
        aria-label="Test"
        data-testid="Test"
        name="Test"
        type="checkbox"
      />
      <label
        for="Test"
      >
        Test Checkbox
      </label>
    </div>
  </div>
</body>
`;
